mapeditor/inspector/towneventswidget.cpp

```diff
@@ -169,9 +169,25 @@ void TownEventsDelegate::setModelData(QWidget * editor, QAbstractItemModel * mod
 	if (auto * ed = qobject_cast<TownEventsWidget *>(editor))
 	{
 		ed->commitChanges(controller);
+		updateModelData(model, index);
 	}
 	else
 	{
 		QStyledItemDelegate::setModelData(editor, model, index);
 	}
 }
+
+void TownEventsDelegate::updateModelData(QAbstractItemModel * model, const QModelIndex & index) const
+{
+	QStringList eventLists;
+	for (const auto & event : town.events)
+	{
+		auto eventName = QString::fromStdString(event.name);
+		eventLists += tr("Day %1 - %2").arg(event.firstOccurrence + 1, 3).arg(eventName);
```

Comment:
what is 3? max 3 digits?

Reply:
Day number will take minimum 3 characters and be right aligned, used same way as on list of events, here in fact it may be not needed